patchy/api.py

```diff
@@ -113,10 +113,10 @@ def _apply_patch(source, patch_text, forwards, name):
             source_file.write(source)
 
         patch_path = os.path.join(tempdir, name + '.patch')
-        with open(patch_path, 'w') as patch_file:
-            patch_file.write(patch_text)
+        with open(patch_path, 'wb') as patch_file:
```

Comment:
I'm not sure this is the correct solution

Reply:
source and patch should be written using the same encoding/line endings configuration.

Reply:
Me neither, I don't have windows environment to work in. But it makes sense that 'patch' only works with unix line endings if it's part of unix environment like mingw?

Reply:
perhaps https://www.gnu.org/software/diffutils/manual/diffutils.html#Changed-White-Space

Reply:
`--ignore-white-space` says it only does spaces and tabs.

I see there is stuff about `diff` stripping trailing cr's but that would apply to patch creation only.

> source and patch should be written using the same encoding/line endings configuration.

Yes they should but the problem here is that patchy is writing the patch out in text mode for which python uses the system default for line endings, rather than using the same line endings in the source file or the file calling patchy.

Probably instead an argument to `patch()` to set `open()`'s `newline` argument would be better. But with @tolomea being the only user I want his feedback on this.